File: shared/src/types/interaction.ts
export enum InteractionType {
  Confirmed = 0,
  ProximityTap = 1,
}

export interface InteractionPostDTO {
  userId: number;
  locationId: number;
  type: InteractionType;
}

export interface InteractionResponseDTO {
  id: number;
  userId: number;
  locationId: number;
  locationName: string;
  interactedAt: string;
  type: InteractionType;
}

export interface UserInteractionStatsDTO {
  userId: number;
  username: string;
  firstName: string;
  lastName: string;
  totalInteractions: number;
}

export interface LocationInteractionStatsDTO {
  locationId: number;
  locationName: string;
  totalInteractions: number;
  latitude: number;
  longitude: number;
}

File: backend/PulseMap/Domain/Interaction.cs
namespace PulseMap.Domain;

public class Interaction
{
    public required int Id { get; set; }
    public required int UserId { get; set; }
    public User? User { get; set; }
    public required int LocationId { get; set; }
    public Location? Location { get; set; }
    public required DateTime InteractedAt { get; set; }
    public required InteractionType Type { get; set; }
}

public enum InteractionType
{
    Confirmed = 0,
    ProximityTap = 1
}
